src/drivers/gps/gps.cpp

```diff
@@ -278,6 +278,7 @@ GPS::~GPS()
 	}
 
 	g_dev = nullptr;
+	orb_unadvertise(_report_gps_pos_pub);
```

Comment:
You should call this from the same thread as the publication is done, ie. in the main loop after `::close(_serial_fd);`